src/components/StringFormatter/__tests__/StringFormatter.spec.js

```diff
@@ -3,34 +3,69 @@
  * @copyright IBM Security 2019
  */
 
-import { shallow } from 'enzyme';
+import { render } from '@testing-library/react';
 import React from 'react';
 
 import { StringFormatter } from '../../..';
-import value from '../_mocks_';
+import { namespace } from '../StringFormatter';
 
 describe('StringFormatter', () => {
-  let stringFormatter;
-
-  beforeEach(() => {
-    stringFormatter = shallow(<StringFormatter value={value} />);
+  test('should have no Axe or DAP violations', async () => {
+    const main = document.createElement('main');
+    render(
+      <StringFormatter
+        value="This is a long test string that would normally be truncated."
+        truncate
+        lines={2}
+        width="50px"
+      />,
+      {
+        // DAP requires a landmark '<main>' in the DOM:
+        container: document.body.appendChild(main),
+      }
+    );
+    await expect(document.body).toHaveNoAxeViolations();
+    await expect(document.body).toHaveNoDAPViolations('StringFormatter');
   });
 
-  it('renders', () => {
-    expect(stringFormatter).toMatchSnapshot();
+  test('should add a custom class', () => {
+    const { getByText } = render(
+      <StringFormatter value="test content" className="custom-class" />
+    );
+    expect(getByText(/test content/i)).toHaveClass('custom-class');
   });
 
-  it('renders the truncated variation', () => {
-    stringFormatter.setProps({ truncate: true });
+  test('should truncate text via the `truncate` prop', () => {
+    const { getAllByText } = render(
+      <StringFormatter value="test content" truncate />
+    );
+    // Expect two elements with the target text: the base component
+    // and the tooltip that is added when `truncate={true}`.
+    expect(getAllByText(/test content/i).length).toBe(2);
+
+    // Expect to see the truncate class on the inner non-tooltip element:
+    expect(getAllByText(/test content/i)[0]).toHaveClass(
+      `${namespace}--truncate`
+    );
+  });
```

Comment:
Not sure how to automate a "visual" test like this (`toBeVisible()` won't apply/work here), so best thing is to check that a tooltip is added & the class name is correct.